server/routes/api/login.js

```diff
@@ -0,0 +1,99 @@
+var express = require('express');
+var router = express.Router();
+var User = require('../../dbSchemas/User');
+
+router.get('/', function(req, res, next){
+    res.render('index', {title : 'Express'});
+});
+
+router.post('/login', function (req, res){
+    var name = req.body.name;
+    var password = req.body.password;
+
+    console.log("login request received");
+
+    User.findOne({name: name, password: password}, function (err, user){
+        if(err){
+            console.log(err);
+            return res.status(500).send();
+        }
+
+        if(!user){
+            return res.status(404).send();
+        }
+
+        // this logs in user
+        req.session.user = user;
+        return res.status(200).send();
+    })
+});
+
+// the following is a test to test to see if you are logged in correctly
+// may be copied and changed as needed
+router.get('/dashboard', function(req, res){
+    if(!req.session.user){
+        console.log("Unuathorized request. Please login.");
+        return res.status(401).send();
+    }  
+
+    return res.status(200).send("Welcome to super-secret API");
+
+});
+
+router.get('/logout', function(req, res){
+    req.session.destroy();
+    return res.status(200).send();
+});
+
+
+router.post('/register', function(req, res){
+    var name = req.body.name;
+    var email = req.body.email;
+    var password = req.body.password;
+    var is_professor = req.body.is_professor;
+    var student_number = req.body.student_number;
```

Comment:
there should really be a conditional check on this attribute because not every user( eg professor) will have a student number but this can be updated in a future pr.